Main.elm

```diff
@@ -0,0 +1,256 @@
+module Main exposing (..)
+
+import Json.Encode as Encode
+import Html exposing (..)
+import Html.Attributes exposing (..)
+import Html.Events exposing (..)
+import Navigation
+import Ports
+import Mastodon
+
+
+type alias Flags =
+    { client : Maybe Mastodon.Client
+    , registration : Maybe Mastodon.AppRegistration
+    }
+
+
+type Msg
+    = NoOp
+    | Register
+    | AppRegistered (Result Mastodon.Error Mastodon.AppRegistration)
+    | AccessToken (Result Mastodon.Error Mastodon.AccessTokenResult)
+    | UserTimeline (Result Mastodon.Error (List Mastodon.Status))
+    | PublicTimeline (Result Mastodon.Error (List Mastodon.Status))
+    | ServerChange String
+    | UrlChange Navigation.Location
```

Comment:
It's maybe a personal preference, but I like to sort my Messages by alphabetical order. When your app grows, it's really easier to look after a specific message.